File: backend/app/schemas/video.py
from __future__ import annotations

from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.metadata import MetadataGenerateResponse


class VideoUploadResponse(BaseModel):
    model_config = ConfigDict(str_strip_whitespace=True)

    video_id: str = Field(description="UUID identifying this upload")
    filename: str = Field(description="Original filename as provided by the client")
    status: Literal["processed"]
    metadata: MetadataGenerateResponse
    content_type: Literal["video", "short"] = "video"
    duration_seconds: Optional[float] = None
    width: Optional[int] = None
    height: Optional[int] = None
    aspect_ratio: Optional[str] = None

File: frontend/types/video.ts
import type { MetadataGenerateResponse } from "@/types/metadata";

export interface VideoUploadResponse {
  video_id: string;
  filename: string;
  status: "processed";
  metadata: MetadataGenerateResponse;
  content_type: "video" | "short";
  duration_seconds: number | null;
  width: number | null;
  height: number | null;
  aspect_ratio: string | null;
}

export type UploadState =
  | "idle"
  | "uploading"
  | "processing"
  | "generating"
  | "complete"
  | "error";
